TestControllers/TestController.psm1

```diff
@@ -151,8 +151,12 @@ Class TestController
 				$CurrentXMLText = Get-Content -Path $file.FullName
 				foreach ($Replace in $this.XMLSecrets.secrets.ReplaceTestXMLStrings.Replace)
 				{
-					$ReplaceString = $Replace.Split("=")[0]
-					$ReplaceWith = $Replace.Split("=")[1]
+					if ($Replace.InnerText) {
+						$Replace.InnerText = [System.Security.SecurityElement]::Escape($Replace.InnerText)
+						$ReplaceString, $ReplaceWith = $Replace.InnerText -split '=',2
+					} else {
```

Comment:
are they always have InnerText? if no InnerText, no need to do this logic of spliting. and no need to do following if ($CurrentXMLText -imatch $ReplaceString)

Reply:
no, some of them have innertext, some of them don't have.